nodejs/lib/inference-session-impl.ts

```diff
@@ -0,0 +1,222 @@
+import {binding, Binding} from './binding';
+import {InferenceSession as InferenceSessionInterface, InferenceSessionFactory} from './inference-session';
+import {OnnxValue} from './onnx-value';
+import {Tensor} from './tensor';
+
+type SessionOptions = InferenceSessionInterface.SessionOptions;
+type RunOptions = InferenceSessionInterface.RunOptions;
+type FeedsType = InferenceSessionInterface.FeedsType;
+type FetchesType = InferenceSessionInterface.FetchesType;
+type ReturnType = InferenceSessionInterface.ReturnType;
+
+class InferenceSession implements InferenceSessionInterface {
+  constructor(session: Binding.InferenceSession) {
+    this.#session = session;
+    // cache metadata
+    this.inputNames = this.#session.inputNames;
+    this.outputNames = this.#session.outputNames;
+  }
+  run(feeds: FeedsType, options?: RunOptions): Promise<ReturnType>;
+  run(feeds: FeedsType, fetches: FetchesType, options?: RunOptions): Promise<ReturnType>;
+  run(feeds: FeedsType, arg1?: FetchesType|RunOptions, arg2?: RunOptions): Promise<ReturnType> {
```

Comment:
EDoes TypeScript automatically map the implementation to the above 2 declarations ?

Reply:
no. Typescript is declaration only and all it does is in compiling time. So I have to write type checks in the impl